### PR TITLE
Add  `count` support for all types

### LINKED ISSUE #5060: [FEA] Support Count on Struct of [ Struct of [String, Map(String,String)], Array(String), Map(String,String) ]
I wish we can support Count on Struct of [ Struct of [String, Map(String,String)], Array(String), Map(String,String) ].

Example:
```
import org.apache.spark.sql.Row
import org.apache.spark.sql.types._

val data = Seq(
    Row("Adam",  Row( Row("address" , Map("state"->"CA")), List("Java","Scala","C++"),  Map("hair"->"black","eye"->"black") )       ),
    Row("Bob",   Row( Row("address" , Map("state"->"NY")), List("Java","Python","C++"), Map("hair"->"red","eye"->"black") )       ),
    Row("Cathy", Row( Row("address" , Map("state"->"CO")), List("Java","C++"),          Map("hair"->"yellow","eye"->"black") )       )
)

val mapType  = DataTypes.createMapType(StringType,StringType)

val schema = new StructType()
                .add("name",StringType)
                .add("prop", new StructType().add("prop2",new StructType()
                                                                .add("propname",StringType)
                                                                .add("propvalue",mapType) 
                                                 )
                                               .add("language", ArrayType(StringType))
                                               .add("bio", mapType)
                     )         

val nestedDF = spark.createDataFrame(spark.sparkContext.parallelize(data),schema)
nestedDF.write.format("parquet").mode("overwrite").save("/tmp/testparquet")
val df1 = spark.read.parquet("/tmp/testparquet")
df1.createOrReplaceTempView("df1")
df1.printSchema

spark.sql("SELECT count(prop) FROM df1").collect()
```

Not-Supported messages:
```
        !Expression <Count> count(prop#364) cannot run on GPU because input expression AttributeReference prop#364 (child StructType(StructField(propname,StringType,true), StructField(propvalue,MapType(StringType,StringType,true),true)) is not supported, child ArrayType(StringType,true) is not supported, child MapType(StringType,StringType,true) is not supported)
```

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>

Close #5060.
The `Count` expression supports all data types now.

### CODE REVIEW

**Comment by HaoYang670** [comment]
build

**Comment by sperlingxx** [comment]
LGTM

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala [hunk 1]
```diff
@@ -2245,9 +2245,7 @@ object GpuOverrides extends Logging {
       ExprChecks.fullAgg(
         TypeSig.LONG, TypeSig.LONG,
         repeatingParamCheck = Some(RepeatingParamCheck(
-          "input", _gpuCommonTypes + TypeSig.DECIMAL_128 +
-              TypeSig.STRUCT.nested(_gpuCommonTypes + TypeSig.DECIMAL_128),
-          TypeSig.all))),
+          "input", TypeSig.all, TypeSig.all))),
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala:2248]: Do we really want to say _every_ time here?  In the past I believe @revans2 had some reservations about stating support for things like UDTs, and there are types we're saying we support here that are not being tested in the integration tests.

Comment (HaoYang670) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala:2248]: Bobby said: we should be able to make this work for any data type. 
https://github.com/NVIDIA/spark-rapids/issues/5060#issuecomment-1079423319

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Review by abellina — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Review by nartal1 — APPROVED** [review]

**Pull request merged**